### ISSUE TITLE
cmd/compile: constant inlining happens even when disabling inlining and optimizations

### ISSUE THREAD

**Issue by dlsniper** [golang/go#35527]
### What version of Go are you using (`go version`)?
go1.13.4

### Does this issue reproduce with the latest release?
I haven't tried with Go 1.14 yet

### What operating system and processor architecture are you using (`go env`)?
windows/amd64

### What did you do?
- use the code at https://github.com/dlsniper/democonstdbg
- use the latest Delve version from master
- place a breakpoint on line 23
- evaluate the following expression:
```
ot == AnotherConst
```

### What did you expect to see?
A boolean result

### What did you see instead?
```
could not find symbol value for AnotherConst
```

**Comment by aarzilli** [comment]
Non-integer constants are never exported to debug_info, it doesn't have anything to do with whether they are inlined or not. Also, I'm not sure it's worth the executable bloat costs, since IIRC there was no way to reuse t

@gopherbot please add label Debugging

**Comment by dlsniper** [comment]
Thanks for the explanation.

The problem was reported to GoLand by a user that had a similar code like the one above.
It's rather common to run a compare operation like this. For example, I use this to setup conditional breakpoints. Also, not everyone will know ahead of time that one is a constant or not when making the comparison, nor suggesting to change the code would be possible.

I'm not sure if https://github.com/go-delve/delve/issues/1753 is related or not.

**Comment by andybons** [comment]
@aclements

**Comment by kzhui125** [comment]
When I debug this code:

https://github.com/adonovan/gopl.io/blob/65c318dde95e3b46a0add461ccf681bd5345b67c/ch1/lissajous/main.go#L69

and hover at the constant variables, the value doesn't appear.

I have to find the definitions what exactly the value is.

**Comment by aarzilli** [comment]
@kzhui125 that doesn't have anything to do with this issue and should be reported to GoLand.

**Comment by kzhui125** [comment]
I am using VSCode, I have reported this issue in vscode-go extension. I was told this is upstream issue. @aarzilli 

https://github.com/microsoft/vscode-go/issues/3226

**Comment by aarzilli** [comment]
You're right, I was confusing this with a different issue.

**Comment by x-yuri** [comment]
I wonder if that's the issue I'm experiencing...

```
$ git clone https://github.com/docker/compose
$ cd compose
$ dlv debug github.com/docker/compose/v2/cmd -- up
Type 'help' for list of commands.
(dlv) b /home/yuri/go/pkg/mod/github.com/compose-spec/compose-go/v2@v2.0.0-beta.2/cli/options.go:260
Breakpoint 1 set at 0x199cc8e for github.com/compose-spec/compose-go/v2/cli.WithEnvFiles.func1() /home/yuri/go/pkg/mod/github.com/compose-spec/compose-go/v2@v2.0.0-beta.2/cli/options.go:260
(dlv) c
> github.com/compose-spec/compose-go/v2/cli.WithEnvFiles.func1() /home/yuri/go/pkg/mod/github.com/compose-spec/compose-go/v2@v2.0.0-beta.2/cli/options.go:260 (hits goroutine(1):1 total:1) (PC: 0x199cc8e)
   255:         return func(o *ProjectOptions) error {
   256:                 if len(file) > 0 {
   257:                         o.EnvFiles = file
   258:                         return nil
   259:                 }
=> 260:                 if v, ok := os.LookupEnv(consts.ComposeDisableDefaultEnvFile); ok {
   261:                         b, err := strconv.ParseBool(v)
   262:                         if err != nil {
   263:                                 return err
   264:                         }
   265:                         if b {
(dlv) p consts.ComposeDisableDefaultEnvFile
Command failed: could not find symbol value for consts
```

Are there any workarounds?

**Comment by x-yuri** [comment]
@aarzilli _(I didn't remember the part about non-integer constants when I was posting the message.)_

Can you possibly explain why non-integer constants are not exported? C for one [doesn't seem to have problems][a] with them.

[a]: https://gist.github.com/x-yuri/748b93ccdf776908ff13a9d2a6cac7ee

**Comment by aarzilli** [comment]
> Can you possibly explain why non-integer constants are not exported?

They just aren't.

**Comment by aclements** [comment]
I've retitled this issue because it's unrelated to inlining. Really this is a feature request for the compiler to include non-integer constants in the debug info. Once we do that, debuggers will be able to access them.
